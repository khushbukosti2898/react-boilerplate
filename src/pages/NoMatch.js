import React from 'react';
// import { Link } from 'react-router';
import ReactHelmet from '../components/common/ReactHelmet';

const NoMatch = () => {
  return (
    <div
      className="error-template text-text-primary 
      d-flex justify-content-center align-items-center"
      data-testid="error"
    >
      <ReactHelmet title="Page Not Found" />
      <div className="text-center">
        <h1 className="text-danger mb-2">Oops!!!</h1>
        <p data-testid="error.message">
          The page you are looking for was not found
        </p>
        <a
          href="/"
          className="btn btn-primary mt-4"
          data-testid="error.home-link"
        >
          Go back to home
        </a>
      </div>
    </div>
  );
};

export default NoMatch;
