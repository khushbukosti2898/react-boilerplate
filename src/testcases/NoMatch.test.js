import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, getByText, render, waitFor } from "@testing-library/react";
import renderer from 'react-test-renderer';

import NoMatch from "../pages/NoMatch";
import NoMatchView from './views/NoMatch-view';

const BASE_URL = `http://localhost`;

const linkClick = (url) => {
    Object.defineProperty(window, "location", {
        value: {
            href: `${BASE_URL}${url}`,
        },
        writable: true,
    });
};

describe("Error component", () => {
    beforeEach(() => {
        global.window = Object.create(window);
        const url = `/temp`;
        linkClick(url);
    });

    it("render no match component", () => {
        const view = renderView();
        expect(view.errorContainer).toBeInTheDocument();
    });

    it("show error message and home page link", () => {
        const error = {
            message: "The page you are looking for was not found",
        };
        const view = renderView();
        expect(view.message).toBeInTheDocument();
        expect(getByText(view.errorContainer, error.message)).toBeInTheDocument();
        expect(view.homeLink).toBeInTheDocument();
    });

    it("on click of home link redirect to home page", async () => {
        const error = {
            message: "The page you are looking for was not found",
        };
        const view = renderView(error);
        const href = view.homeLink.getAttribute("href");
        view.homeLink.onclick = () => {
            linkClick(href);
        };
        expect(window.location.href).toBe("http://localhost/temp");
        await waitFor(() => fireEvent.click(view.homeLink));
        expect(window.location.href).toBe("http://localhost/");
    });
});

describe('Snapshot of no match test', () => {
    test('Snapshot of no match', () => {
        const tree = renderer.create(<NoMatch />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

const renderView = () => {
    const dom = render(<NoMatch />);
    return new NoMatchView(dom);
};