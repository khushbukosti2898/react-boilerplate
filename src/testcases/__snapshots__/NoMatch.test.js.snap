// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of no match test Snapshot of no match 1`] = `
<div
  className="error-template text-text-primary  d-flex justify-content-center align-items-center"
  data-testid="error"
>
  <div
    className="text-center"
  >
    <h1
      className="text-danger mb-2"
    >
      Oops!!!
    </h1>
    <p
      data-testid="error.message"
    >
      The page you are looking for was not found
    </p>
    <a
      className="btn btn-primary mt-4"
      data-testid="error.home-link"
      href="/"
    >
      Go back to home
    </a>
  </div>
</div>
`;
